test/higher_order_ops/test_print.py

```diff
@@ -243,6 +244,99 @@ def f(x):
         self.assertEqual(printed_output, "moo tensor([2])\nmoo tensor([1])")
         self.assertEqual(orig_out, opt_out)
 
+    @skipIfTorchDynamo("Skipped under Dynamo")
+    def test_inductor_python_wrapper_uses_builtin_print(self):
+        """Test that the Python wrapper uses builtins.print instead of HOP for print fallback.
+
+        This verifies that when compiling with inductor (Python wrapper), the generated
+        code uses builtins.print directly rather than calling torch.ops.higher_order.print,
+        which is more efficient and avoids unnecessary overhead.
+        """
+        from torch._inductor.utils import run_and_get_code
+
+        def f(x):
+            torch._higher_order_ops.print("value: {val}", val=x)
+            res = x + x
+            return res
+
+        inputs = (torch.randn(2, 3),)
+
+        # Compile and get the generated code
+        compiled_f = torch.compile(f, backend="inductor")
+        _, codes = run_and_get_code(compiled_f, *inputs)
+
+        # Concatenate all generated code chunks to simplify assertions
+        merged_code = "\n".join(codes)
+
+        # Verify that the merged code uses builtins.print
+        self.assertIn(
+            "builtins.print",
+            merged_code,
+            "Generated code should use builtins.print for print HOP fallback",
+        )
+        # And does not call torch.ops.higher_order.print
+        self.assertNotIn(
+            "torch.ops.higher_order.print",
+            merged_code,
+            "Generated code should not call torch.ops.higher_order.print directly",
+        )
+
+    def test_compile_inductor_cpp_wrapper_print(self):
+        """Test print with C++ wrapper enabled
+
+        C++ wrapper uses std::cout which writes to file descriptor 1 (stdout).
+        We need to redirect the actual file descriptor to capture the output.
+        """
+        import os
+        import tempfile
+
+        from torch._inductor import config
+
+        def f(x):
+            torch._higher_order_ops.print("C++ print test: value={x}", x=x)
+            res = x + x
+            torch._higher_order_ops.print("Result={res}", res=res)
+            return res
+
+        inputs = (torch.randn(2, 3),)
+
+        # Enable C++ wrapper and capture stdout at the file descriptor level
+        with config.patch({"cpp_wrapper": True}):
```

Comment:
can we also check with `run_and_get_code` here?

Reply:
I'll add one!